src/Compilers/Core/Portable/Emit/EmitOptions.cs

```diff
@@ -177,6 +178,14 @@ public bool Equals(EmitOptions other)
                 return false;
             }
 
+            for (int i = 0; i < this.InstrumentationKinds.Length; i++)
```

Comment:
>i < this.InstrumentationKinds.Length [](start = 28, length = 36)

How are we handling different length between this.InstrumentationKinds and other.InstrumentationKinds?